csrc/scheduler/registry_utils.cpp

```diff
@@ -1170,6 +1170,62 @@ bool SchedulerTopologyChecker::hasCyclicReshape(Fusion* fusion) {
   return false;
 }
 
+// Detects incompatible reshape patterns using PERMISSIVE_RESIZE graph.
+// Returns true if rfactor IDs are mapped together (same ValGroup) but have
+// different transformations (different ExprGroups). This indicates the reshape
+// operations cannot be replayed and the fusion must be segmented.
+//
+// Example: slice(tv[36], 0, 24)->tv[24]->reshape([2,3,4]) and
+//          slice(tv[36], 12, 36)->tv[24]->reshape([2,2,6])
+// Both slices produce tv[24] which map together, but reshape differently.
+bool SchedulerTopologyChecker::hasIncompatibleReshapes(Fusion* fusion) {
+  // TODO: Reuse IdModel when possible
+  IdModel id_model(fusion);
+  const auto& permissive_resize_graph = id_model.buildPermissiveResizeGraph();
+
+  for (const ValGroup& val_group :
+       permissive_resize_graph.disjointValSets().disjointSets()) {
+    // Collect all rfactor IDs in this val group
+    // Check for consistency if there are at least 2 rfactor IDs
+    std::vector<IterDomain*> rfactor_ids;
+    for (Val* val : *val_group) {
+      auto id = val->as<IterDomain>();
+      if (id->isRFactorProduct()) {
```

Comment:
I'm wondering if this should be removed. If there's a split or merge at this point, no matter if its part of a reshape or not, we would need to keep it. Think of splits for DID parallelization. It seems all splits and merges here should be considered as something we are not able to modify.

Reply:
> Leads to failures in many multi-device tests. 

What failures did you see?